ansible_runner error with ansible 2.8

Trying the latest testinfra release (3.0.0) with ansible 2.8 (rc2) but running into this issue:

```console
dev-centos-non_container-lvm_osds run-test: commands[7] | py.test -n 8 --durations=0 --sudo -v --connection=ansible --ansible-inventory=/home/jenkins-build/build/workspace/ceph-ansible-prs-dev-centos-non_container-lvm_osds/tests/functional/lvm-osds/hosts /home/jenkins-build/build/workspace/ceph-ansible-prs-dev-centos-non_container-lvm_osds/tests/functional/tests
[2211] /home/jenkins-build/build/workspace/ceph-ansible-prs-dev-centos-non_container-lvm_osds/tests/functional/lvm-osds$ /tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/bin/py.test -n 8 --durations=0 --sudo -v --connection=ansible --ansible-inventory=/home/jenkins-build/build/workspace/ceph-ansible-prs-dev-centos-non_container-lvm_osds/tests/functional/lvm-osds/hosts /home/jenkins-build/build/workspace/ceph-ansible-prs-dev-centos-non_container-lvm_osds/tests/functional/tests
============================= test session starts ==============================
platform linux2 -- Python 2.7.5, pytest-4.4.1, py-1.8.0, pluggy-0.9.0 -- /tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/bin/python
cachedir: /tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/.pytest_cache
rootdir: /home/jenkins-build/build/workspace/ceph-ansible-prs-dev-centos-non_container-lvm_osds/tests, inifile: pytest.ini
plugins: testinfra-3.0.0, xdist-1.28.0, forked-1.0.2
(...)
==================================== ERRORS ====================================
______________ ERROR collecting functional/tests/test_install.py _______________
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/pluggy/hooks.py:289: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/pluggy/manager.py:68: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/pluggy/manager.py:62: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/_pytest/python.py:238: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/_pytest/python.py:414: in _genfunctions
    self.ihook.pytest_generate_tests(metafunc=metafunc)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/pluggy/hooks.py:289: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/pluggy/manager.py:68: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/pluggy/manager.py:62: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/testinfra/plugin.py:113: in pytest_generate_tests
    ansible_inventory=metafunc.config.option.ansible_inventory,
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/testinfra/host.py:143: in get_hosts
    for backend in testinfra.backend.get_backends(hosts, **kwargs):
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/testinfra/backend/__init__.py:91: in get_backends
    for name in klass.get_hosts(host, **kw):
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/testinfra/backend/ansible.py:59: in get_hosts
    return AnsibleRunner.get_runner(inventory).get_hosts(host)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib/python2.7/site-packages/testinfra/utils/ansible_runner.py:124: in get_hosts
    groupmatch = fnmatch.fnmatch(group, pattern)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib64/python2.7/fnmatch.py:43: in fnmatch
    return fnmatchcase(name, pat)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib64/python2.7/fnmatch.py:75: in fnmatchcase
    res = translate(pat)
/tmp/venv.YMrKxhXs5g/dev-centos-non_container-lvm_osds/lib64/python2.7/fnmatch.py:87: in translate
    i, n = 0, len(pat)
E   TypeError: object of type 'NoneType' has no len()
```
I also test 3.0.0 release with ansible 2.7 but the result is the same.

The ansible inventory file is really simple:
```yaml
[mons]
mon0

[osds]
osd0
```

--- a/test/test_backends.py
+++ b/test/test_backends.py
@@ -17,6 +17,7 @@ import os
 import pytest
 import tempfile
 
+import testinfra
 import testinfra.backend
 from testinfra.backend.base import BaseBackend
 from testinfra.backend.base import HostSpec
@@ -173,6 +174,16 @@ def test_ansible_get_host(inventory, expected):
         backend = AnsibleRunner(f.name).get_host('host').backend
         for attr, value in expected.items():
             assert operator.attrgetter(attr)(backend) == value
+
+
+def test_ansible_no_host():
+    with tempfile.NamedTemporaryFile() as f:
+        f.write(b'host\n')
+        f.flush()
+        assert AnsibleRunner(f.name).get_hosts() == ['host']
+        hosts = testinfra.get_hosts(
+            [None], connection='ansible', ansible_inventory=f.name)
+        assert [h.backend.get_pytest_id() for h in hosts] == ['ansible://host']
 
 
 def test_ansible_unhandled_connection():

--- a/testinfra/backend/ansible.py
+++ b/testinfra/backend/ansible.py
@@ -56,4 +56,4 @@ class AnsibleBackend(base.BaseBackend):
     @classmethod
     def get_hosts(cls, host, **kwargs):
         inventory = kwargs.get('ansible_inventory')
-        return AnsibleRunner.get_runner(inventory).get_hosts(host)
+        return AnsibleRunner.get_runner(inventory).get_hosts(host or "all")

--- a/testinfra/utils/ansible_runner.py
+++ b/testinfra/utils/ansible_runner.py
@@ -105,7 +105,7 @@ class AnsibleRunner(object):
         self._host_cache = {}
         super(AnsibleRunner, self).__init__()
 
-    def get_hosts(self, pattern=None):
+    def get_hosts(self, pattern="all"):
         inventory = self.inventory
         result = set()
 

